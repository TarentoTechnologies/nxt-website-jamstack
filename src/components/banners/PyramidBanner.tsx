import { PrimaryCTA } from "../buttons/PrimaryCTA";
import { PrimaryTitle } from "../titles/PrimaryTitle";
import {
  bannerStyles,
  retailInfo,
  imgPadding,
  infoSectionPadding,
} from "./PyramidBanner.module.css";

interface PyramidBannerProps {
  title: string;
  description: string;
  btnDescr?: string;
  pyramidImg: any;
}

export const PyramidBanner = ({
  title,
  description,
  btnDescr,
  pyramidImg,
}: PyramidBannerProps) => {
  return (
    <div className={`container-fluid ${bannerStyles}`}>
      <div className={`row`}>
        <div className={`col-xs-12 col-sm-12 col-md-12 col-lg-4 offset-lg-2`}>
          <div className={`d-flex justify-content-center align-items-center`}>
            <img
              className={`${imgPadding} img-fluid`}
              src={pyramidImg}
              alt="Data pyramid"
              title="Data pyramid"
            />
          </div>
        </div>
        <div className={`col-xs-12 col-sm-12 col-md-12 col-lg-6 col-xl-4`}>
          <div className={`${infoSectionPadding} ${retailInfo}`}>
            <PrimaryTitle title={title} variant="banner" color="white" />
            <p>{description}</p>
            <PrimaryCTA label={btnDescr} primary />
          </div>
        </div>
      </div>
    </div>
  );
};
