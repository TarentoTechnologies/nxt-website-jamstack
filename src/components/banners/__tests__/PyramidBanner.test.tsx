import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { PyramidBanner } from "../PyramidBanner";

const data = {
  PyramidTitle: "Data Pyramid",
  PyramidDescription:
    "Modern businesses are run on information & knowledge. To improve your bottom line and expand your business, you need to have the right information available on time. Fundamentals of good data management is of paramount importance. Make reliable, data driven decisions. Take your business to the next level with our data-platform.",
  PyramidImage: {
    localFile: {
      url: "https://strapi-stage-nxt.tarento.com/uploads/pyramid_52b15649a1.svg",
    },
  },
  PyramidCTAText: "Read More",
  PyramidCTALink: "/data-platform",
};

describe("Test if the PyramidBanner component is working as expected", () => {
  it("<PyramidBanner /> matches snapshot", () => {
    const component = render(
      <PyramidBanner
        title={data.PyramidTitle}
        description={data.PyramidDescription}
        pyramidImg={data.PyramidImage.localFile.url}
        btnDescr={data.PyramidCTAText}
        btnLink={data.PyramidCTALink}
      />
    );
    expect(component.container).toMatchSnapshot();
  });

  it("<PyramidBanner /> renders without exploding", () => {
    const component = render(
      <PyramidBanner
        title={data.PyramidTitle}
        description={data.PyramidDescription}
        pyramidImg={data.PyramidImage.localFile.url}
        btnDescr={data.PyramidCTAText}
        btnLink={data.PyramidCTALink}
      />
    );
    expect(component.getByTestId("pyramid-banner")).toBeInTheDocument();
  });
});
