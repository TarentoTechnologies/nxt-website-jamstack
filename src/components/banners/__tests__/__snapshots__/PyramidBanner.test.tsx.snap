// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the PyramidBanner component is working as expected <PyramidBanner /> matches snapshot 1`] = `
<div>
  <div
    class="container-fluid bannerStyles"
    data-testid="pyramid-banner"
  >
    <div
      class="row"
    >
      <div
        class="col-md-12 col-lg-4 offset-lg-2"
      >
        <div
          class="d-flex justify-content-center align-items-center"
        >
          <img
            alt="Data Pyramid"
            class="imgPadding img-fluid"
            data-aos="fade-up"
            src="https://strapi-stage-nxt.tarento.com/uploads/pyramid_52b15649a1.svg"
            title="Data Pyramid"
          />
        </div>
      </div>
      <div
        class="col-md-12 col-lg-6 col-xl-4"
      >
        <div
          class="infoSectionPadding retailInfo"
          data-aos="fade-up"
        >
          <div
            class="
      undefined
        h1One
        
        
        
        
        
        
        
        
        
        white"
            data-aos="fade-up"
            data-testid="primary-title"
          >
            Data Pyramid
          </div>
          <p>
            Modern businesses are run on information & knowledge. To improve your bottom line and expand your business, you need to have the right information available on time. Fundamentals of good data management is of paramount importance. Make reliable, data driven decisions. Take your business to the next level with our data-platform.
          </p>
          <a
            data-testid="primary-cta"
            href="/data-platform"
          >
            <div
              class="text-center d-flex align-items-center justify-content-center 
      primaryBtn  undefined"
              data-aos="fade-up"
            >
              <label
                class="labelStyles"
              >
                Read More
              </label>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
